src/github/workflows.ts

```diff
@@ -295,7 +299,7 @@ function renderJobs(
       name: step.name,
       id: step.id,
       if: step.if,
-      uses: step.uses,
+      uses: actions.get(step.uses),
```

Comment:
If that's the one reason why `action` is optional, this could easily be changed to:

```suggestion
      uses: step.uses && actions.get(step.uses),
```